From 6afde3005d0afdfab8ca1a183c0089516e5d2f45 Mon Sep 17 00:00:00 2001
From: Shiyong Li <a22381@motorola.com>
Date: Wed, 6 Jan 2016 13:49:07 -0800
Subject: [PATCH 595/959] IKQCOMSD1-3106 arm/dts: Addison: SMD 5.49' OLED panel
 bring up

Add OLED panel support on 8952 based P0 Addison board.

Change-Id: Ibc7261f7db02c1dfd3e47c3285ce2c1df5d4967e
Signed-off-by: zhanggb <zhanggb@motorola.com>
Reviewed-on: http://gerrit.mot.com/810662
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Christopher Fries <cfries@motorola.com>
---
 .../dsi-panel-mot-smd-549-1080p-cmd-common.dtsi    | 139 +++++++++++++++++++++
 .../dts/qcom/dsi-panel-mot-smd-549-1080p-cmd.dtsi  |  17 +++
 arch/arm/boot/dts/qcom/msm8952-addison.dtsi        |  31 +++--
 3 files changed, 177 insertions(+), 10 deletions(-)
 create mode 100644 arch/arm/boot/dts/qcom/dsi-panel-mot-smd-549-1080p-cmd-common.dtsi
 create mode 100644 arch/arm/boot/dts/qcom/dsi-panel-mot-smd-549-1080p-cmd.dtsi

diff --git a/arch/arm/boot/dts/qcom/dsi-panel-mot-smd-549-1080p-cmd-common.dtsi b/arch/arm/boot/dts/qcom/dsi-panel-mot-smd-549-1080p-cmd-common.dtsi
new file mode 100644
index 00000000000..02c9206c54b
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/dsi-panel-mot-smd-549-1080p-cmd-common.dtsi
@@ -0,0 +1,139 @@
+/* Copyright (c) 2016, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+compatible = "qcom,mdss-dsi-panel";
+qcom,mdss-dsi-panel-name = "mipi_mot_cmd_smd_1080p_549_v0";
+qcom,mdss-dsi-panel-supplier = "smd";
+qcom,mdss-dsi-panel-type = "dsi_cmd_mode";
+qcom,mdss-dsi-panel-framerate = <60>;
+qcom,mdss-dsi-virtual-channel-id = <0>;
+qcom,mdss-dsi-stream = <0>;
+qcom,mdss-dsi-panel-width = <1080>;
+qcom,mdss-dsi-panel-height = <1920>;
+/* To update dimension based on final spec */
+qcom,mdss-pan-physical-width-dimension = <65>;
+qcom,mdss-pan-physical-height-dimension = <115>;
+
+/* To update clock rate */
+qcom,mdss-dsi-panel-clockrate = <0>;
+qcom,mdss-dsi-h-front-porch = <44>;
+qcom,mdss-dsi-h-back-porch = <32>;
+qcom,mdss-dsi-h-pulse-width = <12>;
+qcom,mdss-dsi-h-sync-skew = <0>;
+qcom,mdss-dsi-v-back-porch = <3>;
+qcom,mdss-dsi-v-front-porch = <9>;
+qcom,mdss-dsi-v-pulse-width = <4>;
+qcom,mdss-dsi-h-left-border = <0>;
+qcom,mdss-dsi-h-right-border = <0>;
+qcom,mdss-dsi-v-top-border = <0>;
+qcom,mdss-dsi-v-bottom-border = <0>;
+qcom,mdss-dsi-bpp = <24>;
+qcom,mdss-dsi-underflow-color = <0xff>;
+qcom,mdss-dsi-border-color = <0>;
+qcom,mdss-dsi-on-command = [
+	05 01 00 00 78 00 01 11
+	29 01 00 00 00 00 03 fc 5a 5a
+	29 01 00 00 00 00 03 f4 00 01
+	15 01 00 00 00 00 02 57 00
+	29 01 00 00 00 00 03 fc a5 a5
+	15 01 00 00 00 00 02 53 20
+	15 01 00 00 00 00 02 55 03
+	15 01 00 00 00 00 02 51 00
+	05 01 00 00 00 00 01 29];	/* display on */
+qcom,mdss-dsi-off-command = [
+	05 01 00 00 0a 00 01 28		/* display_off */
+	05 01 00 00 55 00 01 10];	/* enter sleep mode */
+
+qcom,mdss-dsi-on-command-state = "dsi_lp_mode";
+qcom,mdss-dsi-off-command-state = "dsi_lp_mode";
+qcom,mdss-dsi-h-sync-pulse = <0>;
+qcom,mdss-dsi-traffic-mode = "burst_mode";
+qcom,mdss-dsi-bllp-eof-power-mode;
+qcom,mdss-dsi-bllp-power-mode;
+qcom,mdss-dsi-lane-0-state;
+qcom,mdss-dsi-lane-1-state;
+qcom,mdss-dsi-lane-2-state;
+qcom,mdss-dsi-lane-3-state;
+/* To caculate timing with final clock rate setting */
+qcom,mdss-dsi-panel-timings = [b0 23 1b 00 94 93 1e 25 15 03 04 00];
+qcom,mdss-dsi-t-clk-post = <0x1c>;
+qcom,mdss-dsi-t-clk-pre = <0x2c>;
+qcom,mdss-dsi-bl-min-level = <1>;
+qcom,mdss-dsi-bl-max-level = <255>;
+qcom,mdss-dsi-dma-trigger = "trigger_sw";
+qcom,mdss-dsi-mdp-trigger = "none";
+qcom,mdss-dsi-bl-pmic-control-type = "bl_ctrl_dcs";
+qcom,mdss-dsi-reset-sequence = <1 11>;
+qcom,mdss-dsi-lp11-init;
+qcom,mdss-dsi-tx-eot-append;
+
+qcom,mdss-dsi-te-pin-select = <1>;
+qcom,mdss-dsi-te-dcs-command = <1>;
+qcom,mdss-dsi-te-check-enable;
+qcom,mdss-dsi-te-using-te-pin;
+
+qcom,panel-supply-entries {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	qcom,panel-supply-entry@0 {
+		reg = <0>;
+		qcom,supply-name = "vci_disp";
+		qcom,supply-min-voltage = <3000000>;
+		qcom,supply-max-voltage = <3000000>;
+		qcom,supply-enable-load = <300000>;
+		qcom,supply-disable-load = <100>;
+		qcom,supply-pre-on-sleep = <0>;
+		qcom,supply-post-on-sleep = <1>;
+		qcom,supply-pre-off-sleep = <0>;
+		qcom,supply-post-off-sleep = <1>;
+	}
+;
+	qcom,panel-supply-entry@1 {
+		reg = <1>;
+		qcom,supply-name = "vio_disp";
+		qcom,supply-min-voltage = <1800000>;
+		qcom,supply-max-voltage = <1800000>;
+		qcom,supply-enable-load = <300000>;
+		qcom,supply-disable-load = <100>;
+		qcom,supply-pre-on-sleep = <0>;
+		qcom,supply-post-on-sleep = <3>;
+		qcom,supply-pre-off-sleep = <0>;
+		qcom,supply-post-off-sleep = <1>;
+	};
+
+	qcom,panel-supply-entry@2 {
+		reg = <2>;
+		qcom,supply-name = "elvdd_disp";
+		qcom,supply-min-voltage = <4600000>;
+		qcom,supply-max-voltage = <4600000>;
+		qcom,supply-enable-load = <100000>;
+		qcom,supply-disable-load = <100>;
+		qcom,supply-pre-on-sleep = <10>;
+		qcom,supply-post-on-sleep = <10>;
+		qcom,supply-pre-off-sleep = <10>;
+		qcom,supply-post-off-sleep = <10>;
+	};
+
+	qcom,panel-supply-entry@3 {
+		reg = <3>;
+		qcom,supply-name = "elvss_disp";
+		qcom,supply-min-voltage = <5000000>;
+		qcom,supply-max-voltage = <5000000>;
+		qcom,supply-enable-load = <100000>;
+		qcom,supply-disable-load = <100>;
+		qcom,supply-pre-on-sleep = <10>;
+		qcom,supply-post-on-sleep = <10>;
+		qcom,supply-pre-off-sleep = <10>;
+		qcom,supply-post-off-sleep = <10>;
+	};
+};
diff --git a/arch/arm/boot/dts/qcom/dsi-panel-mot-smd-549-1080p-cmd.dtsi b/arch/arm/boot/dts/qcom/dsi-panel-mot-smd-549-1080p-cmd.dtsi
new file mode 100644
index 00000000000..6c37cef7514
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/dsi-panel-mot-smd-549-1080p-cmd.dtsi
@@ -0,0 +1,17 @@
+/* Copyright (c) 2016, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+&mdss_mdp {
+	mot_smd_549_1080p_cmd_v0: qcom,mdss_dsi_mot_smd_549_1080p_cmd_v0 {
+		#include "dsi-panel-mot-smd-549-1080p-cmd-common.dtsi"
+	};
+};
diff --git a/arch/arm/boot/dts/qcom/msm8952-addison.dtsi b/arch/arm/boot/dts/qcom/msm8952-addison.dtsi
index 25a95173e4e..9f56d10bcc0 100644
--- a/arch/arm/boot/dts/qcom/msm8952-addison.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-addison.dtsi
@@ -12,8 +12,7 @@
 
 
 #include "msm8952-moto-common.dtsi"
-#include "dsi-panel-mot-lgd-546-1080p-cmd.dtsi"
-#include "dsi-panel-mot-boe-546-1080p-cmd.dtsi"
+#include "dsi-panel-mot-smd-549-1080p-cmd.dtsi"
 
 / {
 	model = "Addison";
@@ -22,30 +21,42 @@
 };
 
 &soc {
-	vddio_disp_vreg: vddio_disp_vreg {
+	vci_disp_vreg: vci_disp_vreg {
 		compatible = "regulator-fixed";
-		regulator-name = "vddio_disp";
+		regulator-name = "vci_disp";
+		gpio = <&msm_gpio 32 0>;
+		startup-delay-us = <300>;
+		enable-active-high;
+		regulator-boot-on;
+	};
+
+	vio_disp_vreg: vio_disp_vreg {
+		compatible = "regulator-fixed";
+		regulator-name = "vio_disp";
 		vin-supply = <&pm8950_l6>;
+		gpio = <&msm_gpio 33 0>;
 		startup-delay-us = <300>;
+		enable-active-high;
 		regulator-boot-on;
-		regulator-always-on;
 	};
 };
 
 &mdss_dsi0 {
-	qcom,dsi-pref-prim-pan = <&mot_lgd_546_1080p_cmd_v0>;
-	vddio_disp-supply = <&vddio_disp_vreg>;
-	vsp_disp-supply = <&lab_regulator>;
-	vsn_disp-supply = <&ibb_regulator>;
+	qcom,dsi-pref-prim-pan = <&mot_smd_549_1080p_cmd_v0>;
+	vci_disp-supply = <&vci_disp_vreg>;
+	vio_disp-supply = <&vio_disp_vreg>;
+	elvdd_disp-supply = <&lab_regulator>;
+	elvss_disp-supply = <&ibb_regulator>;
 };
 
 /* display boost and inverting boost regulators */
 &labibb {
 	status = "ok";
-	qpnp,qpnp-labibb-mode = "lcd";
+	qpnp,qpnp-labibb-mode = "amoled";
 };
 
 &ibb_regulator {
+	qcom,qpnp-ibb-init-amoled-voltage = <5000000>;
 	qcom,qpnp-ibb-discharge-resistor = <32>;
 };
 
-- 
2.11.0

